### PR TITLE
Korablev Nikita. Lab №1. Option №4.

### PR DESCRIPTION
**Pull request by NikitaKorablev**
A plugin has been implemented that renames identifiers (variable name, class name, function name).
Example:
```c++
class A {
public:
    A() {};
    ~A();
};
void func() {
    A* a = new A;
    A b;
    delete a;
};
```
Result:
```c++
class B {
public:
    B() {};
    ~B();
};
void func() {
    B* a = new B;
    B b;
    delete a;
 };
```

### CODE REVIEW

File: clang/labs/lab1/korablev_nikita/RenameIdentificator.cpp [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+#include "clang/ASTMatchers/ASTMatchFinder.h"
+#include "clang/ASTMatchers/ASTMatchers.h"
+#include "clang/Tooling/Transformer/RewriteRule.h"
+#include "clang/Tooling/Transformer/Transformer.h"
+#include "clang/Tooling/Transformer/Stencil.h"
+
+using namespace clang;
+using namespace clang::ast_matchers;
+using namespace clang::transformer;
```
Comment (aobolensk) [inline clang/labs/lab1/korablev_nikita/RenameIdentificator.cpp]: Please, avoid using "using namespace *". "Using" statement for particular namespace members is ok (like ast_matchers and transformer)

File: clang/labs/lab1/korablev_nikita/RenameIdentificator.cpp [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+        if (VD->getType().getAsString() == oldName + " *") {
+            rewriter.ReplaceText(VD->getTypeSourceInfo()->getTypeLoc().getBeginLoc(), name.size(), newName);
+            rewriter.overwriteChangedFiles();
+        }
+
+        if (VD->getType().getAsString() == oldName) {
+            rewriter.ReplaceText(VD->getTypeSourceInfo()->getTypeLoc().getBeginLoc(), name.size(), newName);
+            rewriter.overwriteChangedFiles();
+        }
```
Comment (aobolensk) [inline clang/labs/lab1/korablev_nikita/RenameIdentificator.cpp]: It looks like these two ifs can be merged

**Review by aobolensk — CHANGES_REQUESTED** [review]

**Comment by m-ly4** [comment]
Could you please align code with a help of clang-format tool?
And also fix warnings printed by https://github.com/NN-complr-tech/llvm-nnsu-2024/actions/runs/8294689254/job/22700181439?pr=8

**Comment by aobolensk** [comment]
Please, update the branch to go through updated clang-format check

**Review by m-ly4 — APPROVED** [review]

**Review by aobolensk — APPROVED** [review]

**Pull request merged**